include/clink/linalg/sparse_vector.h

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2021 The Clink Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CLINK_LINALG_SPARSE_VECTOR_H_
+#define CLINK_LINALG_SPARSE_VECTOR_H_
+
+#include "llvm/ADT/SmallVector.h"
+
+namespace clink {
+
+// A sparse vector of double values.
+class SparseVector {
+public:
+  // Constructor for SparseVector.
+  // `n` stands for the number of dimensions of the vector.
+  SparseVector(const int n);
+
+  // Move operations are supported.
+  SparseVector(SparseVector &&other);
+  SparseVector &operator=(SparseVector &&other);
+
+  // This class is not copyable or assignable.
+  SparseVector(const SparseVector &other) = delete;
+  SparseVector &operator=(const SparseVector &) = delete;
+
+  // Sets the value at a certain index of the vector.
+  void set(int index, double value);
+
+  // Gets the value at a certain index of the vector.
+  double get(int index);
```

Comment:
Can we add the const keyword for this method?

Same for size().

Reply:
OK. I'll make the change.